DOC: Contributor guideline should guide contributors for expected local mypy failures

#### Location of the documentation

[Location](https://pandas.pydata.org/pandas-docs/stable/development/contributing.html#style-guidelines).

#### Documentation problem

See conversation [here](https://gitter.im/pydata/pandas?at=5eee0e2ba813c72dcffccbb5).

#### Suggested fix for documentation

This 'failing' of mypy should be noted so that inexperienced contributors don't have to Google around and experienced ones don't have to explain repeatedly.

I came across this in [this context](https://github.com/pandas-dev/pandas/pull/32542#discussion_r442959117).

> Another option would be to only run mypy before a push instead of on every commit. This might be achievable with specifying the stage in the config
> 
> https://pre-commit.com/

Not sure that work would - see below

```bash
(pandas-dev) marco@marco-Predator-PH315-52:~/pandas-dev$ git fetch upstream master
From https://github.com/pandas-dev/pandas
 * branch                master     -> FETCH_HEAD
(pandas-dev) marco@marco-Predator-PH315-52:~/pandas-dev$ git reset --hard upstream/master
HEAD is now at 0159cba6e CLN: dont consolidate in indexing (#34679)
(pandas-dev) marco@marco-Predator-PH315-52:~/pandas-dev$ pre-commit run --all
black....................................................................Passed
flake8...................................................................Passed
flake8-pyx...............................................................Passed
flake8-pxd...............................................................Passed
isort....................................................................Failed
- hook id: isort
- files were modified by this hook

Fixing /home/marco/pandas-dev/web/pandas_web.py

mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

pandas/core/dtypes/generic.py:10: error: unused 'type: ignore' comment
pandas/core/resample.py:970: error: unused 'type: ignore' comment
Found 2 errors in 2 files (checked 1043 source files)

(pandas-dev) marco@marco-Predator-PH315-52:~/pandas-dev$ mypy pandas
Success: no issues found in 1024 source files
```

IMHO it could be removed from `pre-commit` - if you're adding type annotations you'll need to run `mypy pandas` anyway, and if you're not then its false positives are a distraction (especially for new contributors)

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -30,15 +30,3 @@ repos:
     -   id: isort
         language: python_venv
         exclude: ^pandas/__init__\.py$|^pandas/core/api\.py$
--   repo: https://github.com/pre-commit/mirrors-mypy
-    rev: v0.730
-    hooks:
-     -  id: mypy
-        args:
-          # As long as a some files are excluded from check-untyped-defs
-          # we have to exclude it from the pre-commit hook as the configuration
-          # is based on modules but the hook runs on files.
-          - --no-check-untyped-defs
-          - --follow-imports
-          - skip
-        files: pandas/

